wire-gradle-plugin/src/test/kotlin/com/squareup/wire/gradle/WirePluginTest.kt

```diff
@@ -0,0 +1,343 @@
+    val fixtureRoot = File("src/test/projects/different-java-out")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .contains("Writing com.squareup.geology.Period")
+        .contains("src/test/projects/different-java-out/build/custom")
+  }
+
+  @Test
+  fun differentKotlinOutputDir() {
+    val fixtureRoot = File("src/test/projects/different-kotlin-out")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .contains("Writing com.squareup.geology.Period")
+        .contains("src/test/projects/different-kotlin-out/build/custom")
+  }
+
+  @Test
+  fun kotlinTargetMissingKotlinPlugin() {
+    val fixtureRoot = File("src/test/projects/missing-kotlin-plugin")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .buildAndFail()
+
+    assertThat(result.task(":generateProtos")).isNull()
+    assertThat(result.output)
+        .contains(
+            "Task with name 'compileKotlin' not found in root project 'missing-kotlin-plugin'"
+        )
+  }
+
+  @Test
+  fun rootKeepsField() {
+    val fixtureRoot = File("src/test/projects/field-root")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .doesNotContain("Writing com.squareup.geology.Period")
+
+    val generatedProto =
+      File(fixtureRoot, "build/generated/src/main/java/com/squareup/dinosaurs/Dinosaur.kt")
+    assertThat(generatedProto).exists()
+
+    assertThat(generatedProto.readText())
+        .contains(
+            """
+            |data class Dinosaur(
+            |    /**
+            |     * Common name of this dinosaur, like "Stegosaurus".
+            |     */
+            |    @field:WireField(tag = 1, adapter = "com.squareup.wire.ProtoAdapter#STRING") val name: String? =
+            |            null,
+            |    val unknownFields: ByteString = ByteString.EMPTY
+            |) : Message<Dinosaur, Dinosaur.Builder>(ADAPTER, unknownFields) {
+            """.trimMargin()
+        )
+  }
+
+  @Test
+  fun pruneRemovesField() {
+    val fixtureRoot = File("src/test/projects/field-prune")
+
+    val result = gradleRunner
+        .withProjectDir(fixtureRoot)
+        .build()
+
+    assertThat(result.task(":generateProtos")).isNotNull
+    assertThat(result.output)
+        .contains("Writing com.squareup.dinosaurs.Dinosaur")
+        .contains("Writing com.squareup.geology.Period")
+
+    val generatedProto =
+      File(fixtureRoot, "build/generated/src/main/java/com/squareup/dinosaurs/Dinosaur.kt")
+    assertThat(generatedProto).exists()
+
+    assertThat(generatedProto.readText())
+        .doesNotContain(
```

Comment:
I think this is too brittle, if the structure or the formatting of code emitted by the compiler changes this test will start passing even if the field is present. Would be good to figure out a way to make it more robust.

Reply:
Def agree, but landed on this after concluding that parsing Kotlin into an AST for assertions didn't seem possible.  Google's compile-testing library only supports Java at the moment.

Reply:
I'd go with just reducing the string to something that's more robust, e.g. just the property name or whatever makes sense.

Reply:
cool, updated!